### PR TITLE
Always set the Cmd for source Docker image

### PR DESCRIPTION
**Pull request by mfojtik**
@csrwng PTAL :)

Fixes: #8376

### CODE REVIEW

File: pkg/build/builder/source.go [hunk 1]
```diff
@@ -326,6 +326,9 @@ func extractSourceFromImage(dockerClient DockerClient, image, buildDir string, i
 	container, err := dockerClient.CreateContainer(docker.CreateContainerOptions{
 		Config: &docker.Config{
 			Image: image,
+			// Cmd must be set for images that does not contain the CMD or ENTRYPOINT
```
Comment (csrwng) [inline pkg/build/builder/source.go]: s/does not/do not/

**Comment by csrwng** [comment]
really minor nit, otherwise LGTM

**Comment by jorgemoralespou** [comment]
@mfojtik @csrwng Is it safe to set a "/bin/sh" command on any image. I'm using scratch as FROM and it seems to work, but wondering if any other image might break.

**Comment by csrwng** [comment]
@jorgemoralespou the command is never run, so I would think it's safe

**Comment by csrwng** [comment]
We only create the container, extract files from it and delete it

**Comment by mfojtik** [comment]
@csrwng fixed, thanks :-)

**Comment by mfojtik** [comment]
[merge]

**Comment by mfojtik** [comment]
[testonlyextended][extended:core(build can have Docker image source)]

File: pkg/build/builder/source.go [hunk 2]
```diff
@@ -319,15 +319,20 @@ func extractSourceFromImage(dockerClient DockerClient, image, buildDir string, i
 		if err := dockerClient.PullImage(docker.PullImageOptions{Repository: image}, dockerAuth); err != nil {
 			return fmt.Errorf("error pulling image %v: %v", image, err)
 		}
+	}
 
+	containerConfig := &docker.Config{Image: image}
+	if inspect, err := dockerClient.InspectImage(image); err != nil {
+		return err
+	} else {
+		// In case the Docker image do not specify the entrypoint
```
Comment (csrwng) [inline pkg/build/builder/source.go]: s/do not/does not/

Comment (mfojtik) [inline pkg/build/builder/source.go]: lol :-) thanks

**Comment by bparees** [comment]
low risk, has approval pending successful extended test run.

**Comment by bparees** [comment]
tests passed, approved for [merge]

**Comment by mfojtik** [comment]
[merge]

**Comment by mfojtik** [comment]
[merge] one more time

**Comment by jorgemoralespou** [comment]
@mfojtik is our ci system located in Spain and at siesta time? Feels consistent.

**Pull request merged**